cloud-controller-manager/do/metrics.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2020 DigitalOcean
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package do
+
+import "github.com/prometheus/client_golang/prometheus"
+
+type metrics struct {
+	host               string
+	apiRequestDuration *prometheus.HistogramVec
+	runLoopDuration    *prometheus.HistogramVec
+	reconcileDuration  *prometheus.HistogramVec
+}
+
+// create metrics
+var (
+	apiRequestDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: "firewall",
+			Name:      "http_request_duration_seconds",
+			Help:      "This is a histogram for tracking the duration of firewall API requests.",
+			Buckets:   prometheus.ExponentialBuckets(0.01, 2, 14),
+		},
+		[]string{"method", "code"},
+	)
+	runLoopDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: "firewall",
+			Name:      "run_loop_duration_seconds",
+			Help:      "This is a histogram for tracking the duration of the run loop and whether it succeeds or not.",
+			Buckets:   prometheus.ExponentialBuckets(0.01, 2, 15),
+		},
+		[]string{"success"},
+	)
+	reconcileDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: "firewall",
+			Name:      "reconcile_duration_seconds",
+			Help:      "The duration of time, in seconds, that it takes for the firewall reconcile to run and ensure that the firewall is reconciled.",
+			Buckets:   prometheus.ExponentialBuckets(0.01, 2, 14),
+		},
+		[]string{"reconcile_type"},
```

Comment:
Is this the reconcile origin mentioned by Timo in one of his comments?

> The ability to break down reconciles by origin sounds useful to me as well.

Reply:
No, this was an update based on his earlier comment from today.
> This should be a single label name (such as reconcile_type) with four different label values to ease filtering and aggregating.

Reply:
I interpreted as there being 4 possible values (5 if you include a default `"none"` value), 'event_add', 'event_update', 'event_delete' and 'firewall_loop'.